ingenialink/utils/_utils.py

```diff
@@ -339,3 +341,66 @@ def set_logger_level(level):
 
     """
     lib.set_log_level(level)
+
+
+def convert_bytes_to_dtype(data, dtype):
+    """Convert data in bytes to corresponding dtype."""
+    if dtype in [REG_DTYPE.S8,
+                 REG_DTYPE.S16,
+                 REG_DTYPE.S32]:
+        value = int.from_bytes(
+            data,
+            "little",
+            signed=True
+        )
+    elif dtype == REG_DTYPE.FLOAT:
+        if len(data) > 4:
+            data = data[:4]
+        [value] = struct.unpack('f',
+                                data
+                                )
+    elif dtype == REG_DTYPE.STR:
+        value = data.decode("utf-8")
+    else:
+        value = int.from_bytes(
+            data,
+            "little"
+        )
+    return value
+
+
+def convert_dtype_to_bytes(data, dtype):
+    """Convert data in dtype to bytes.
+    Args:
+        data: Data to convert.
+        dtype (REG_DTYPE): Data type.
+    """
+    # auto cast floats if register is not float
+    if dtype == REG_DTYPE.FLOAT:
+        data = float(data)
+    elif dtype != REG_DTYPE.DOMAIN:
+        data = int(data)
+    if dtype == REG_DTYPE.FLOAT:
+        data = struct.pack('f', data)
+    elif dtype != REG_DTYPE.DOMAIN:
+        bytes_length = 2
+        signed = False
+        if dtype == REG_DTYPE.U8:
+            bytes_length = 1
+        elif dtype == REG_DTYPE.S8:
+            bytes_length = 1
+            signed = True
+        elif dtype == REG_DTYPE.U16:
+            bytes_length = 2
+        elif dtype == REG_DTYPE.S16:
+            bytes_length = 2
+            signed = True
+        elif dtype == REG_DTYPE.U32:
+            bytes_length = 4
+        elif dtype == REG_DTYPE.S32:
+            bytes_length = 4
+            signed = True
+        data = data.to_bytes(bytes_length,
+                             byteorder='little',
+                             signed=signed)
```

Comment:
Maybe you can write this function more clear, like:
``` python
if dtype == REG_DTYPE.DOMAIN:
    return data
if dtype == REG_DTYPE.FLOAT:
    return data = struct.pack('f', float(data))
__dtype_value = {
    REG_DTYPE.U8: (1, False),
    REG_DTYPE.S8: (1, True)
    REG_DTYPE.U16: (2, False),
    REG_DTYPE.S16: (2, True),
    REG_DTYPE.U32: (4, False),
    REG_DTYPE.S32: (4, True)
}
bytes_length, signed = __dtype_value[dtype]
...
```
Or something like this, what do you think?

Reply:
In this case, with str type will raise an exception.

Reply:
I like it, I will change it.